presto-main/src/main/java/com/facebook/presto/operator/EnforceUniqueColumnsOperator.java

```diff
@@ -0,0 +1,197 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.RunLengthEncodedBlock;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.sql.planner.plan.PlanNodeId;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class EnforceUniqueColumnsOperator
+        implements Operator
+{
+    private static final long VALUES_PER_REQUEST = 1 << 20;
+
+    public static class UuidOperatorFactory
+            implements OperatorFactory
+    {
+        private final int operatorId;
+        private final PlanNodeId planNodeId;
+        private final List<Type> types;
+        private boolean closed;
+        private final AtomicLong valuePool = new AtomicLong();
+
+        public UuidOperatorFactory(
+                int operatorId,
+                PlanNodeId planNodeId,
+                List<? extends Type> types)
+        {
+            this.operatorId = operatorId;
+            this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.types = ImmutableList.copyOf(types);
+        }
+
+        @Override
+        public List<Type> getTypes()
+        {
+            return types;
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            checkState(!closed, "Factory is already closed");
+
+            OperatorContext operatorContext = driverContext.addOperatorContext(
+                    operatorId,
+                    planNodeId,
+                    EnforceUniqueColumnsOperator.class.getSimpleName());
+            return new EnforceUniqueColumnsOperator(operatorContext, types, valuePool);
+        }
+
+        @Override
+        public void close()
+        {
+            closed = true;
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            return new UuidOperatorFactory(operatorId, planNodeId, types);
+        }
+    }
+
+    private final OperatorContext operatorContext;
+    private boolean finishing;
+    private final int taskId;
```

Comment:
I am going to leverage taskId and stageId to make sure that rows from two different stages for `AssigneUniqueId` are unique.